test/dhtest.c

```diff
@@ -195,12 +201,18 @@ int main(int argc, char *argv[])
  err:
     ERR_print_errors_fp(stderr);
 
-    OPENSSL_free(abuf);
-    OPENSSL_free(bbuf);
-    DH_free(b);
-    DH_free(a);
-    BN_GENCB_free(_cb);
-    BIO_free(out);
+    if (abuf)
+        OPENSSL_free(abuf);
+    if (bbuf)
+        OPENSSL_free(bbuf);
+    if (b)
+        DH_free(b);
+    if (a)
+        DH_free(a);
+    if (_cb)
+        BN_GENCB_free(_cb);
+    if (out)
+        BIO_free(out);
```

Comment:
Just like free(), the OpenSSL freeing routines are no-ops if the given parameter is NULL.  Therefore, these particular conditions are not needed (or if they are, you've found a bug in some of OpenSSL's freeing routines; please report if that's the case)